cibuildwheel/__main__.py

```diff
@@ -154,22 +164,38 @@ def main():
         build_selector=build_selector,
         repair_command=repair_command,
         environment=environment,
+        dependency_constraints=dependency_constraints,
     )
 
     if platform == 'linux':
         manylinux_x86_64_image = os.environ.get('CIBW_MANYLINUX_X86_64_IMAGE', 'manylinux2010')
         manylinux_i686_image = os.environ.get('CIBW_MANYLINUX_I686_IMAGE', 'manylinux2010')
 
-        default_manylinux_images_x86_64 = {'manylinux1': 'quay.io/pypa/manylinux1_x86_64',
-                                           'manylinux2010': 'quay.io/pypa/manylinux2010_x86_64',
-                                           'manylinux2014': 'quay.io/pypa/manylinux2014_x86_64'}
-        default_manylinux_images_i686 = {'manylinux1': 'quay.io/pypa/manylinux1_i686',
-                                         'manylinux2010': 'quay.io/pypa/manylinux2010_i686',
-                                         'manylinux2014': 'quay.io/pypa/manylinux2014_i686'}
+        pinned_docker_images_file = os.path.join(
+            os.path.dirname(__file__), 'resources', 'pinned_docker_images.cfg'
+        )
+        pinned_docker_images = ConfigParser()
+        pinned_docker_images.read(pinned_docker_images_file)
```

Comment:
Also, is there a reason it's better/makes more sense to have manylinux version as main level, and architecture as secondary, rather than the opposite way as it was?

Reply:
I suppose I preferred it this way around in the config file, since the x86_64 image and the i686 image had the same version tag and are next to each other.

So this:

```ini
[manylinux1]
x86_64 = quay.io/pypa/manylinux1_x86_64:2020-01-31-d8fa357
i686 = quay.io/pypa/manylinux1_i686:2020-01-31-d8fa357

[manylinux2010]
x86_64 = quay.io/pypa/manylinux2010_x86_64:2020-01-31-046f791
i686 = quay.io/pypa/manylinux2010_i686:2020-01-31-046f791

[manylinux2014]
x86_64 = quay.io/pypa/manylinux2014_x86_64:2020-01-31-3350900
i686 = quay.io/pypa/manylinux2014_i686:2020-01-31-3350900
```

rather than this:

```ini
[x86_64]
manylinux1 = quay.io/pypa/manylinux1_x86_64:2020-01-31-d8fa357
manylinux2010 = quay.io/pypa/manylinux2010_x86_64:2020-01-31-046f791
manylinux2014 = quay.io/pypa/manylinux2014_x86_64:2020-01-31-3350900

[i686]
manylinux1 = quay.io/pypa/manylinux1_i686:2020-01-31-d8fa357
manylinux2010 = quay.io/pypa/manylinux2010_i686:2020-01-31-046f791
manylinux2014 = quay.io/pypa/manylinux2014_i686:2020-01-31-3350900
```

I'm less convinced of this argument as I write it out. Which do you prefer (taking into account the code in __main__.py as well)?

Reply:
I see. I was mainly curious why; not sure there's a 'right' way. The only argument I can think of is that we should again take PyPy and the new architectures from #269 into account: these don't have `manylinux1`, `manylinux2010`, and `manylinux2014`. So intuitively, I would list them per implementation/architecture, rather than per manylinux version, but that might very well just be because I'm used to that.

Reply:
I see, so you'd have different images available depending on implementation/architecture.

something like:

```ini
[cp_x86_64]
manylinux1 = quay.io/pypa/manylinux1_x86_64:2020-01-31-d8fa357
manylinux2010 = quay.io/pypa/manylinux2010_x86_64:2020-01-31-046f791
manylinux2014 = quay.io/pypa/manylinux2014_x86_64:2020-01-31-3350900

[cp_i686]
manylinux1 = quay.io/pypa/manylinux1_i686:2020-01-31-d8fa357
manylinux2010 = quay.io/pypa/manylinux2010_i686:2020-01-31-046f791
manylinux2014 = quay.io/pypa/manylinux2014_i686:2020-01-31-3350900

[pp_x86_64]
manylinux2010 = some other image
```

over...

```
[manylinux1]
cp_x86_64 = quay.io/pypa/manylinux1_x86_64:2020-01-31-d8fa357
cp_i686 = quay.io/pypa/manylinux1_i686:2020-01-31-d8fa357

[manylinux2010]
cp_x86_64 = quay.io/pypa/manylinux2010_x86_64:2020-01-31-046f791
cp_i686 = quay.io/pypa/manylinux2010_i686:2020-01-31-046f791
pp_x86_64 = some other image

[manylinux2014]
cp_x86_64 = quay.io/pypa/manylinux2014_x86_64:2020-01-31-3350900
cp_i686 = quay.io/pypa/manylinux2014_i686:2020-01-31-3350900
```

Yeah, I think you way is neater

Reply:
Something tells me the repeated `if`s from below might also be slightly shorter then (cfr. the `.get(...) or ...` from before?)

Reply:
Yes, that will clean up much nicer with this structure too :)